### PR TITLE
Added BatchSourceContext.setInput(InputFormatProvider)

### PR DESCRIPTION
**Pull request by chtyim**
- Remove usage of BatchContext.getHadoopJob
- Also move MapReduce classes under "mapreduce" package

### CODE REVIEW

File: cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/etl/batch/source/TimePartitionedFileSetDatasetAvroSource.java [hunk 1]
```diff
@@ -83,11 +88,21 @@ protected void addFileSetProperties(FileSetProperties.Builder properties) {
...
+      Configuration hConf = job.getConfiguration();
+      hConf.clear();
+
+      Schema avroSchema = new Schema.Parser().parse(tpfsAvroConfig.schema);
+      AvroJob.setInputKeySchema(job, avroSchema);
+      for (Map.Entry<String, String> entry : hConf) {
+        config.put(entry.getKey(), entry.getValue());
+      }
+    } catch (IOException e) {
+      // Shouldn't happen
```
Comment (gokulavasan) [inline cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/etl/batch/source/TimePartitionedFileSetDatasetAvroSource.java]: Shouldn't happen because ``tpfsAvroConfig.schema`` has been validated before in ``configurePipeline`` method?

Comment (chtyim) [inline cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/etl/batch/source/TimePartitionedFileSetDatasetAvroSource.java]: The IOException is from the `Job.getInstance()` call.

File: cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/etl/common/db/ETLDBInputFormat.java [hunk 2]
```diff
@@ -40,6 +41,15 @@
   private Driver driver;
   private JDBCDriverShim driverShim;
 
+  public static void setInput(Configuration conf,
```
Comment (gokulavasan) [inline cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/etl/common/db/ETLDBInputFormat.java]: Is this used anywhere?

File: cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/etl/batch/source/TimePartitionedFileSetDatasetAvroSource.java (hunk 1 above)
Comment (gokulavasan) [inline cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/etl/batch/source/TimePartitionedFileSetDatasetAvroSource.java]: Oh ok. :+1:

**Comment by gokulavasan** [comment]
One question. Rest LGTM :+1:

File: cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/etl/common/db/ETLDBInputFormat.java (hunk 2 above)
Comment (chtyim) [inline cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/etl/common/db/ETLDBInputFormat.java]: Oh. Not anymore. It was used by DBSource during refactoring.

File: cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/etl/batch/source/DBSource.java [hunk 3]
```diff
@@ -77,14 +79,16 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
...
+  public void prepareRun(BatchSourceContext context) throws Exception {
     LOG.debug("pluginType = {}; pluginName = {}; connectionString = {}; importQuery = {}; " +
                 "countQuery = {}",
               dbSourceConfig.jdbcPluginType, dbSourceConfig.jdbcPluginName,
               dbSourceConfig.connectionString, dbSourceConfig.importQuery, dbSourceConfig.countQuery);
 
-    Job job = context.getHadoopJob();
+    Job job = Job.getInstance();
     Configuration hConf = job.getConfiguration();
+    hConf.clear();
```
Comment (bdmogal) [inline cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/etl/batch/source/DBSource.java]: why is this needed? So the ``hConf`` passed to ``SourceInputFormatProvider`` on line 101 does not have anything other than ``DBConfiguration``, ``ETLDBInputFormat`` properties?

**Comment by chtyim** [comment]
Build passed: http://builds.cask.co/browse/CDAP-DUT3117-3

File: cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/etl/batch/source/DBSource.java (hunk 3 above)
Comment (chtyim) [inline cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/etl/batch/source/DBSource.java]: otherwise it'll have everything from the `core-default.xml`, which are unnecessary.

**Comment by chtyim** [comment]
Squashed commits. Merging.
